apps/server/src/services/integration-service/OscIntegration.ts

```diff
@@ -113,6 +110,44 @@ export class OscIntegration implements IIntegration<OscSubscription, OSCSettings
     }
   }
 
+  private stringToOSCArgs(argsString): ArgumentType[] {
+
+    // NOTE: regex taken from https://stackoverflow.com/questions/4031900/split-a-string-by-whitespace-keeping-quoted-segments-allowing-escaped-quotes
+    const parseRegex = /[\w.]+|"(?:\\"|[^"])+"/g;
```

Comment:
Nice, could we extract this regex and add some simple tests to it?

Reply:
we have a regex directory in the client and utils.
We just make simple test for the happy paths to provide documentation

See as example
https://github.com/cpvalente/ontime/blob/master/apps/client/src/common/utils/regex.ts

And the tests
https://github.com/cpvalente/ontime/blob/master/apps/client/src/common/utils/__tests__/regex.test.ts

Reply:
Nice. Let me know if you need help

Right - wrong might be all we need for now. 
The tests are a good place to show a few examples of expected payloads and something we don't handle at all.

If there are bug reports in future that we track down to this regex we can leverage the existing test to expose the bug

Reply:
For this regex the application isn't so much interested in the regex pass/failing as it is the result of the regex groups. This regex will turn a string like `'1.1 "string with spaces" 3'` into an array `[1.1, "string with spaces", 3]` so the tests to add would more make sure it does that properly. I'll take a crack at the tests if alex doesn't get around to it by the time I am free this evening.

Reply:
I'll leave this to @alex-Arc as the string parsing logic is now completely different then what I had originally committed so they will be able to write tests for what they are intending to parse.